### PR TITLE
Fix InputFieldsAndArgumentsOfCorrectLength validation rule

### PR DESCRIPTION
**Pull request by Shane32**
See #3301

### CODE REVIEW

File: src/GraphQL/Validation/Rules.Custom/InputFieldsAndArgumentsOfCorrectLength.cs [hunk 1]
```diff
@@ -59,7 +59,7 @@ public override ValueTask VisitFieldAsync(ValidationContext context, GraphQLVari
             new MatchingNodeVisitor<GraphQLArgument>((arg, context) => CheckLength(arg, arg.Value, context.TypeInfo.GetArgument(), context)),
             new MatchingNodeVisitor<GraphQLObjectField>((field, context) =>
             {
-                if (context.TypeInfo.GetInputType(1) is IInputObjectGraphType input)
+                if (context.TypeInfo.GetInputType(1)?.GetNamedType() is IInputObjectGraphType input)
```
Comment (sungam3r) [inline src/GraphQL/Validation/Rules.Custom/InputFieldsAndArgumentsOfCorrectLength.cs:62]: 🤦‍♂️

File: src/GraphQL.Tests/Validation/InputFieldsAndArgumentsOfCorrectLengthRequiredTests.cs [hunk 2]
```diff
@@ -0,0 +1,358 @@
+using GraphQL.Validation.Rules;
+
+namespace GraphQL.Tests.Validation;
+
+public class InputFieldsAndArgumentsOfCorrectLengthRequiredTests : ValidationTestBase<InputFieldsAndArgumentsOfCorrectLength, ValidationSchema>
```
Comment (sungam3r) [inline src/GraphQL.Tests/Validation/InputFieldsAndArgumentsOfCorrectLengthRequiredTests.cs:5]: You don't want to put tests in InputFieldsAndArgumentsOfCorrectLengthTests.cs?

**Review by sungam3r — APPROVED** [review]

File: src/GraphQL.Tests/Validation/InputFieldsAndArgumentsOfCorrectLengthRequiredTests.cs (hunk 2 above)
Comment (Shane32) [reply src/GraphQL.Tests/Validation/InputFieldsAndArgumentsOfCorrectLengthRequiredTests.cs:5]: Either way is fine with me.  It just seemed easier at the time to copy the whole file.  Go ahead and make any changes to the PR you wish.

Comment (sungam3r) [reply src/GraphQL.Tests/Validation/InputFieldsAndArgumentsOfCorrectLengthRequiredTests.cs:5]: It seemed to me we had tests and I really found a file with tests. The next question that I asked myself - how they differ? Only `non-null` type modifier?

Comment (Shane32) [reply src/GraphQL.Tests/Validation/InputFieldsAndArgumentsOfCorrectLengthRequiredTests.cs:5]: Right; the only difference is that it is testing fields with non-null modifiers.  The tests are all the same, I think.

Comment (sungam3r) [reply src/GraphQL.Tests/Validation/InputFieldsAndArgumentsOfCorrectLengthRequiredTests.cs:5]: ok

**Pull request merged**